apps/web/src/features/safenet/components/new-safe/SafenetReview/index.tsx

```diff
@@ -0,0 +1,151 @@
+import SafenetLogo from '@/public/images/safenet/logo-safenet.svg'
+import { gtmSetChainId } from '@/services/analytics/gtm'
+import { useAppDispatch } from '@/store'
+import { useLazyDeploySafenetAccountQuery } from '@/store/safenet'
+import type { ReplayedSafeProps } from '@/store/slices'
+import CheckIcon from '@mui/icons-material/Check'
+import { Box, Typography } from '@mui/material'
+import { EMPTY_DATA, ZERO_ADDRESS } from '@safe-global/protocol-kit/dist/src/utils/constants'
+import { useRouter } from 'next/router'
+import type { Dispatch, SetStateAction } from 'react'
+import { useEffect } from 'react'
+import css from './styles.module.css'
+
+export type UseSubmitSafenetReviewHandlerProps = {
+  data: NewSafeFormData
+  onSubmit: (data: Partial<NewSafeFormData>) => void
+  setSubmitError: Dispatch<SetStateAction<string | undefined>>
+  setIsCreating: Dispatch<SetStateAction<boolean>>
+}
+
+export type SafenetReviewType = {
+  handleCreateSafeClick: () => void
+}
+
+export const useSubmitSafenetReviewHandler = ({
+  data,
+  onSubmit,
+  setSubmitError,
+  setIsCreating,
+}: UseSubmitSafenetReviewHandlerProps): SafenetReviewType => {
+  const router = useRouter()
+  const dispatch = useAppDispatch()
+  const [deploy, { data: deploymentData, error }] = useLazyDeploySafenetAccountQuery()
+
+  useEffect(() => {
+    if (!error) return
+    console.error(error)
+    setSubmitError('Error creating the Safe Account. Please try again later.')
+    setIsCreating(false)
+  }, [error, setSubmitError, setIsCreating])
+
+  useEffect(() => {
+    if (!deploymentData || !data) return
+
+    const replayedSafeWithNonce: ReplayedSafeProps = {
+      factoryAddress: deploymentData.factoryAddress,
+      masterCopy: deploymentData.masterCopy,
+      safeAccountConfig: {
+        threshold: deploymentData.safeAccountConfig.threshold,
+        owners: data.owners.map((owner) => owner.address),
+        to: deploymentData.safeAccountConfig.to || ZERO_ADDRESS,
+        data: deploymentData.safeAccountConfig.data || EMPTY_DATA,
+        fallbackHandler: deploymentData.safeAccountConfig.fallbackHandler || ZERO_ADDRESS,
+        paymentReceiver: ZERO_ADDRESS, // data.paymentReceiver ?? ECOSYSTEM_ID_ADDRESS,
+      },
+      saltNonce: deploymentData.saltNonce,
+      safeVersion: deploymentData.safeVersion,
+    }
+
+    data.networks.map((chain) => {
+      gtmSetChainId(chain.chainId)
+
+      replayCounterfactualSafeDeployment(
+        chain.chainId,
+        deploymentData.safeAddress,
+        replayedSafeWithNonce,
+        data.name,
+        dispatch,
+        PayMethod.PayNow,
+      )
+
+      safeCreationDispatch(SafeCreationEvent.RELAYING, {
+        groupKey: CF_TX_GROUP_KEY,
+        safeAddress: deploymentData.safeAddress,
+        chainId: chain.chainId,
+      })
+
+      //trackEvent({ ...OVERVIEW_EVENTS.PROCEED_WITH_TX, label: 'deployment', category: CREATE_SAFE_CATEGORY })
```

Comment:
We should create a new github issue regarding analytics events for Safenet, but can leave it out of scope in the PR.

Reply:
https://github.com/safe-global/safenet/issues/369